File: .tmp-debug/check_db.mjs
import mysql from "mysql2/promise";

const conn = await mysql.createConnection({
  uri: "mysql://u144001284_caprina:Capitan@123456@lavender-armadillo-743548.hostingersite.com:3306/u144001284_caprina",
  connectTimeout: 15000,
});

console.log("== status distribution (all, not deleted) ==");
let [rows] = await conn.query("SELECT status, COUNT(*) c FROM shipments WHERE deleted_at IS NULL GROUP BY status ORDER BY c DESC");
console.log(rows);

console.log("== total ==");
[rows] = await conn.query("SELECT COUNT(*) c FROM shipments WHERE deleted_at IS NULL");
console.log(rows);

console.log("== date range ==");
[rows] = await conn.query("SELECT MIN(created_at) mn, MAX(created_at) mx FROM shipments WHERE deleted_at IS NULL");
console.log(rows);

console.log("== last 7 days count ==");
[rows] = await conn.query("SELECT COUNT(*) c FROM shipments WHERE deleted_at IS NULL AND created_at >= NOW() - INTERVAL 7 DAY");
console.log(rows);

console.log("== last 7 days status distribution ==");
[rows] = await conn.query("SELECT status, COUNT(*) c FROM shipments WHERE deleted_at IS NULL AND created_at >= NOW() - INTERVAL 7 DAY GROUP BY status ORDER BY c DESC");
console.log(rows);

await conn.end();

File: .tmp-debug/check_db.py
import sys, io
sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')

try:
    import pymysql
except ImportError:
    import subprocess
    subprocess.run([sys.executable, "-m", "pip", "install", "pymysql", "--break-system-packages", "--quiet"])
    import pymysql

conn = pymysql.connect(
    host="lavender-armadillo-743548.hostingersite.com",
    user="u144001284_caprina",
    password="Capitan@123456",
    database="u144001284_caprina",
    port=3306,
    connect_timeout=15,
)
cur = conn.cursor()

print("== status distribution (all shipments, not deleted) ==")
cur.execute("SELECT status, COUNT(*) FROM shipments WHERE deleted_at IS NULL GROUP BY status ORDER BY COUNT(*) DESC")
for row in cur.fetchall():
    print(row)

print()
print("== total shipments ==")
cur.execute("SELECT COUNT(*) FROM shipments WHERE deleted_at IS NULL")
print(cur.fetchone())

print()
print("== date range of created_at ==")
cur.execute("SELECT MIN(created_at), MAX(created_at) FROM shipments WHERE deleted_at IS NULL")
print(cur.fetchone())

print()
print("== shipments in last 7 days ==")
cur.execute("SELECT COUNT(*) FROM shipments WHERE deleted_at IS NULL AND created_at >= NOW() - INTERVAL 7 DAY")
print(cur.fetchone())

print()
print("== status distribution in last 7 days ==")
cur.execute("SELECT status, COUNT(*) FROM shipments WHERE deleted_at IS NULL AND created_at >= NOW() - INTERVAL 7 DAY GROUP BY status ORDER BY COUNT(*) DESC")
for row in cur.fetchall():
    print(row)

cur.close()
conn.close()
